src/aioquic/h3/connection.py

```diff
@@ -1042,23 +1099,23 @@ def _receive_stream_data_uni(
 
     def _validate_settings(self, settings: Dict[int, int]) -> None:
         if Setting.H3_DATAGRAM in settings:
-            if settings[Setting.H3_DATAGRAM] not in (0, 1):
-                raise SettingsError("H3_DATAGRAM setting must be 0 or 1")
+            if settings[Setting.H3_DATAGRAM] not in {0, 1, 2}:
```

Comment:
The change between draft-00 and draft-04 was the setting identifier, not the value. The value is still 0 or 1

Reply:
Thank you, I misunderstood it. I'll revert the change(s).